src/main/java/games/strategy/triplea/oddsCalculator/ta/AggregateResults.java

```diff
@@ -13,14 +14,17 @@
 import games.strategy.util.IntegerMap;
 import games.strategy.util.Tuple;
 
+/**
+ * Results from odds calculator over multiple runs.
+ */
 public class AggregateResults implements Serializable {
   private static final long serialVersionUID = -556699626060414738L;
   // can be empty!
   private final List<BattleResults> m_results;
   private long m_time;
 
   public AggregateResults(final int expectedCount) {
-    m_results = new ArrayList<>(expectedCount);
+    m_results = Collections.synchronizedList(new ArrayList<>(expectedCount));
```

Comment:
I _think_ this is sufficient to ensure there are no thread safety issues in `AggregateResults`.  My brief analysis suggests that once `m_results` is populated by the calculation thread(s), it is never again modified, so that the various `getXXX()` methods in this class do not need to worry about the state of `m_results` possibly changing while they are executing (if it could change, even while synchronized, many of the `getXXX()` methods would then not be atomic, and could yield incorrect results).

If it's true that `m_results` is never again modified once the simulation runs are complete, `AggregateResults` could be made immutable and the thread safety discussion would be moot.  What if each task created by `ConcurrentOddsCalculator` simply pushed its `BattleResults` to a thread-safe collection that was local to `calculate()`?  Then when all tasks have finished, a new `AggregateResults` could be created from that collection with an immutable non-thread-safe copy?  The following incomplete diff captures what I'm thinking:

```diff
diff --git a/src/main/java/games/strategy/triplea/oddsCalculator/ta/AggregateResults.java b/src/main/java/games/strategy/triplea/oddsCalculator/ta/AggregateResults.java
index 03d0cc1..6df890d 100644
--- a/src/main/java/games/strategy/triplea/oddsCalculator/ta/AggregateResults.java
+++ b/src/main/java/games/strategy/triplea/oddsCalculator/ta/AggregateResults.java
@@ -23,6 +23,11 @@ public class AggregateResults implements Serializable {
   private final List<BattleResults> m_results;
   private long m_time;
 
+  public AggregateResults(final long time, final Collection<BattleResults> results) {
+    m_time = time;
+    m_results = new ArrayList<>(results);
+  }
+
   public AggregateResults(final int expectedCount) {
     m_results = Collections.synchronizedList(new ArrayList<>(expectedCount));
   }
diff --git a/src/main/java/games/strategy/triplea/oddsCalculator/ta/ConcurrentOddsCalculator.java b/src/main/java/games/strategy/triplea/oddsCalculator/ta/ConcurrentOddsCalculator.java
index cdba242..861d329 100644
--- a/src/main/java/games/strategy/triplea/oddsCalculator/ta/ConcurrentOddsCalculator.java
+++ b/src/main/java/games/strategy/triplea/oddsCalculator/ta/ConcurrentOddsCalculator.java
@@ -21,14 +23,14 @@ public class ConcurrentOddsCalculator extends OddsCalculator {
   public AggregateResults calculate(final OddsCalculatorParameters parameters) {
     final ExecutorService executor = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors());
     final long start = System.currentTimeMillis();
-    final AggregateResults aggregateResults = new AggregateResults(parameters.runCount);
+    final Queue<BattleResults> battleResults = new ConcurrentLinkedQueue<>();
 
     final int totalRunCount = parameters.runCount;
     parameters.setRunCount(1);
     for (int i = 0; i < totalRunCount; i++) {
       executor.submit(() -> {
         if (!cancelled) {
-          aggregateResults.addResult(OddsCalculator.doSimulation(parameters));
+          battleResults.add(OddsCalculator.doSimulation(parameters));
         }
       });
     }
@@ -40,8 +42,8 @@ public class ConcurrentOddsCalculator extends OddsCalculator {
     } catch (final InterruptedException e) {
       Thread.currentThread().interrupt();
     }
-    aggregateResults.setTime(System.currentTimeMillis() - start);
-    return aggregateResults;
+
+    return new AggregateResults(System.currentTimeMillis() - start, battleResults);
   }
 
   /*
```

It's kind of six of one, half-dozen of another.  I'm not sure it's worth it, but it does keep all the threading concerns local to `ConcurrentOddsCalculator`.

<hr>

**NOTE:** Another thing that needs to be considered with the use of `Collections#synchronizedList()` is that if an instance of `AggregateResults` is deserialized from an older version, the resulting `m_results` will not be synchronized.  Not sure if that's a problem.  I'm assuming a deserialized `AggregateResults` would never be accessed concurrently.

<hr>

**NOTE:** It appears the following methods in this class are now unused:

* `addResults()`
* `getResults()`

Reply:
@DanVanAtta I submitted a PR to your branch that more completely fleshes-out the `AggregateResults` immutability proposal for your consideration.